include/clap/ext/draft/scratch-memory.h

```diff
@@ -18,11 +34,11 @@ typedef struct clap_host_scratch_memory {
    // return "false".
    //
    // Note that any memory the host allocates to satisfy
-   // the requested scratch size can be de-allocated
-   // when the plugin is de-activated.
+   // the requested scratch size can be de-allocated by the
+   // host when the plugin is de-activated.
    //
    // [main-thread & being-activated]
-   bool(CLAP_ABI *request_size)(const clap_host_t *host, size_t scratch_size_bytes);
+   bool(CLAP_ABI *pre_reserve)(const clap_host_t *host, size_t scratch_size_bytes);
```

Comment:
Maybe just call it `reserve`?

This function is not as easy as it seems to impl, as for returning true, it implies that the memory is successfully reserved. For bitwig, this means, allocating the memory for each threads in the thread pool, sending the new memory to those threads, in a way which guarantees that the memory will be setup before the plugin is processed.

Reply:
`reserve` is good!

Yeah, I think this extension will be tricky for hosts that process plugins across multiple threads, but for single-plugin-thread hosts I don't think it will be too complicated. I imagine this extension will be most appealing to hosts running on memory-constrained platforms, and I would guess most of those will have a single plugin thread. At least that's my use case at the moment :).

Reply:
I wonder if it wouldn't be better if the host was pulling the scratch size :thinking: